### PR TITLE
EWPP-874: Add Link field to the Text with featured media paragraph.

### PR DESCRIPTION
**Pull request by yenyasinn**

### CODE REVIEW

File: modules/oe_paragraphs_media/config/install/paragraphs.paragraphs_type.oe_text_feature_media.yml [hunk 1]
```diff
@@ -4,5 +4,6 @@ dependencies: {  }
 id: oe_text_feature_media
 label: 'Text with Featured media'
 icon_uuid: null
+icon_default: null
```
Comment (yenyasinn) [inline modules/oe_paragraphs_media/config/install/paragraphs.paragraphs_type.oe_text_feature_media.yml:7]: Drupal added it after fresh installation.

File: modules/oe_paragraphs_media/oe_paragraphs_media.post_update.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+function oe_paragraphs_media_post_update_00001(array &$sandbox) {
+  $storage = new FileStorage(drupal_get_path('module', 'oe_paragraphs_media') . '/config/post_updates/00001_add_link_field');
+
+  $field_config = 'field.field.paragraph.oe_text_feature_media.field_oe_link';
+  $config_record = $storage->read($field_config);
+  $field = FieldConfig::load($config_record['id']);
+  if ($field) {
+    // Bail out if the field already exists.
+    return t('The field_oe_link already exists.');
+  }
```
Comment (nagyad) [inline modules/oe_paragraphs_media/oe_paragraphs_media.post_update.php:25]: I'm not sure this is needed we know that the field doesn't exist at this point.

Comment (yenyasinn) [reply modules/oe_paragraphs_media/oe_paragraphs_media.post_update.php:25]: I've just copy pasted from the module this template...
In any case it is good that we provide information about an issue if it happens.

**Review by nagyad — DISMISSED** [review]

File: modules/oe_paragraphs_media/config/install/core.entity_form_display.paragraph.oe_text_feature_media.default.yml [hunk 3]
```diff
@@ -2,25 +2,35 @@ langcode: en
 status: true
 dependencies:
   config:
+    - field.field.paragraph.oe_text_feature_media.field_oe_link
     - field.field.paragraph.oe_text_feature_media.field_oe_media
     - field.field.paragraph.oe_text_feature_media.field_oe_plain_text_long
     - field.field.paragraph.oe_text_feature_media.field_oe_text_long
     - field.field.paragraph.oe_text_feature_media.field_oe_title
-    - image.style.thumbnail
```
Comment (22Alexandra) [inline modules/oe_paragraphs_media/config/install/core.entity_form_display.paragraph.oe_text_feature_media.default.yml:9]: Why was this removed?

File: modules/oe_paragraphs_media/config/install/core.entity_view_display.paragraph.oe_text_feature_media.default.yml [hunk 4]
```diff
@@ -2,19 +2,32 @@ langcode: en
...
 dependencies:
   config:
+    - field.field.paragraph.oe_text_feature_media.field_oe_link
     - field.field.paragraph.oe_text_feature_media.field_oe_media
     - field.field.paragraph.oe_text_feature_media.field_oe_plain_text_long
     - field.field.paragraph.oe_text_feature_media.field_oe_text_long
     - field.field.paragraph.oe_text_feature_media.field_oe_title
     - paragraphs.paragraphs_type.oe_text_feature_media
   module:
-    - image
```
Comment (22Alexandra) [inline modules/oe_paragraphs_media/config/install/core.entity_view_display.paragraph.oe_text_feature_media.default.yml:11]: Why removed?

File: modules/oe_paragraphs_media/config/install/field.field.paragraph.oe_text_feature_media.field_oe_link.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+bundle: oe_text_feature_media
+label: Link
+description: ''
+required: false
+translatable: true
+default_value: {  }
+default_value_callback: ''
+settings:
+  link_type: 17
+  title: 1
```
Comment (22Alexandra) [inline modules/oe_paragraphs_media/config/install/field.field.paragraph.oe_text_feature_media.field_oe_link.yml:21]: We should set the title required when the link is filled in.

**Review by 22Alexandra — CHANGES_REQUESTED** [review]

File: modules/oe_paragraphs_media/config/install/core.entity_form_display.paragraph.oe_text_feature_media.default.yml (hunk 3 above)
Comment (yenyasinn) [reply modules/oe_paragraphs_media/config/install/core.entity_form_display.paragraph.oe_text_feature_media.default.yml:9]: Because it isn't used. Drupal removed it after fresh site installation.

File: modules/oe_paragraphs_media/config/install/core.entity_view_display.paragraph.oe_text_feature_media.default.yml (hunk 4 above)
Comment (yenyasinn) [reply modules/oe_paragraphs_media/config/install/core.entity_view_display.paragraph.oe_text_feature_media.default.yml:11]: Because it isn't used. Drupal removed it after fresh site installation.

File: modules/oe_paragraphs_media/config/install/field.field.paragraph.oe_text_feature_media.field_oe_link.yml (hunk 5 above)
Comment (yenyasinn) [reply modules/oe_paragraphs_media/config/install/field.field.paragraph.oe_text_feature_media.field_oe_link.yml]: It isn't defined in the specification but for consistency with usage of Link field in other paragraphs it was set.

**Review by 22Alexandra — APPROVED** [review]

**Review by nagyad — APPROVED** [review]
